### PR TITLE
Simplify faker offsets and limits

### LINKED ISSUE #27802: source-faker crash
```
Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 13, in <module></module>
    launch(source, sys.argv[1:])
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 185, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 115, in run
    yield from map(AirbyteEntrypoint.airbyte_message_to_string, self.read(source_spec, config, config_catalog, state))
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 155, in read
    yield from self.source.read(self.logger, config, catalog, state)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 141, in read
    raise e
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 121, in read
    yield from self._read_stream(
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 196, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 265, in _read_incremental
    for message_counter, record_data_or_message in enumerate(records, start=1):
  File "/airbyte/integration_code/source_faker/streams.py", line 64, in read_records
    updated_at = product["updated_at"]
KeyError: 'updated_at'
```

From https://cloud.airbyte.com/workspaces/b61bc266-ef3c-460c-af2f-f70da4a2993c/connections/ba2e9aae-2159-451a-a089-cab65cea7e72/job-history

### PR DESCRIPTION
**Pull request by evantahler**
Fixes a bug introduced in https://github.com/airbytehq/airbyte/pull/27684 in which products could have a null updated_at.  This PR also greatly simplifies the generation limits for each stream.

Closes https://github.com/airbytehq/airbyte/issues/27802

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`fcf3f163a3`](https://github.com/airbytehq/airbyte/pull/27807/commits/fcf3f163a35fa9cffa57592ede39cdd4075d1569)) - ✅

⏲️  Total pipeline duration: 146 seconds

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/evan_faker-update_at-fix/1687979796/fcf3f163a35fa9cffa57592ede39cdd4075d1569/source-faker/3.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`fcf3f163a3`](https://github.com/airbytehq/airbyte/pull/27807/commits/fcf3f163a35fa9cffa57592ede39cdd4075d1569)) - ✅

⏲️  Total pipeline duration: 131 seconds

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/evan_faker-update_at-fix/1687980151/fcf3f163a35fa9cffa57592ede39cdd4075d1569/source-faker/3.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

File: airbyte-integrations/connectors/source-faker/source_faker/streams.py [hunk 1]
```diff
@@ -47,23 +48,20 @@ def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
...
-            yield generate_estimate(self.name, rows_to_emit, median_record_byte_size)
+        rows_to_emit = len(products)
+        yield generate_estimate(self.name, rows_to_emit, median_record_byte_size)
 
         for product in products:
-            if product["id"] > total_records:
-                product["updated_at"] = format_airbyte_time(datetime.datetime.now())
+            if product["id"] <= self.count:
+                updated_at = format_airbyte_time(datetime.datetime.now())
+                product["updated_at"] = updated_at
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-faker/source_faker/streams.py:61]: This is the explicit fix - updated_at is always set to a value now

**Review by bnchrch — APPROVED** [review]
Approved!

I will be honest, there is a lot in this connector thats hard to follow.

But its a connector for testing not for users 😛

**Comment by evantahler** [comment]
It's the multi threading... and it was my first-ever Python project 😅  

It's probably due for a refactor!
